src/aio/aio_theme_changer.py

```diff
@@ -53,142 +65,97 @@ def __init__(
-                {
-                    "label": str(i),
-                    "label_id": "theme-switch-label",
-                    "value": dbc_themes_url[i],
-                }
-                for i in dbc_themes_url
-            ]
-            # assign id to dark themes in order to apply css
-            for option in radio_props["options"]:
-                if option["label"].lower() in dbc_dark_themes:
-                    option["label_id"] = "theme-switch-label-dark"
-
-        button_props = button_props.copy()
-        if "children" not in button_props:
-            button_props["children"] = "Change Theme"
-        if "color" not in button_props:
-            button_props["color"] = "secondary"
-        if "outline" not in button_props:
-            button_props["outline"] = True
-        if "size" not in button_props:
-            button_props["size"] = "sm"
-
-        offcanvas_props = offcanvas_props.copy()
-        if "children" not in offcanvas_props:
-            offcanvas_props["children"] = [
-                dbc.RadioItems(id=self.ids.radio(aio_id), **radio_props),
-            ]
-        if "title" not in offcanvas_props:
-            offcanvas_props["title"] = "Select a Theme"
-        if "is_open" not in offcanvas_props:
-            offcanvas_props["is_open"] = False
-        if "style" not in offcanvas_props:
-            offcanvas_props["style"] = {"width": 235}
-
-        super().__init__(
-            [
-                dbc.Button(id=self.ids.button(aio_id), **button_props),
-                dbc.Offcanvas(id=self.ids.offcanvas(aio_id), **offcanvas_props),
-                html.Div(
-                    id=self.ids.dummy_div(aio_id),
-                    children=radio_props["value"],
-                    hidden=True,
+        # concat custom themes and bootstrap themes
+        themes_url = (custom_themes | dbc_themes_url) if custom_themes else dbc_themes_url
+        # concat custom dark themes and bootstrap dark themes
+        dark_themes = (dbc_dark_themes + custom_dark_themes) if custom_dark_themes else dbc_dark_themes
+        dark_themes_url = [url for theme, url in themes_url.items() if theme in dark_themes]
+
+        # init button_props
+        if button_props is None:
+            button_props = {}
+        # set default params if they don't exist
+        button_props.setdefault("children", "Change Theme")
+        button_props.setdefault("color", "secondary")
+        button_props.setdefault("outline", True)
+        button_props.setdefault("size", "sm")
+
+        # init radio_props
+        if radio_props is None:
+            radio_props = {}
+        # set default params if they don't exist
+        radio_props.setdefault("options", [{'label': k, 'value': v} for k, v in themes_url.items()])
+        radio_props.setdefault("value", dbc.themes.BOOTSTRAP)
+        # style the labels
+        radio_props['options'] = [
+            {
+                "label": html.Div(
```

Comment:
This is a much better way to do it, but dbc.RadioItems only supported components as props for labels in dbc v1.4.0 released Feb 23.  It will break for anyone using an older version of dbc

Reply:
Last commit adds backward compatibility with dbc 1.3.1
Rollback using `#theme-switch-label`/`#theme-switch-label-dark` CSS styles

But still no need to apply `label_id` when user provide a custom `radio_props['options']`, like
```python
ThemeChangerAIO(
    aio_id="theme",
    radio_props={
        "options": [
            {"label": "Cyborg", "value": dbc.themes.CYBORG},
            {"label": "Vapor", "value": dbc.themes.VAPOR}
        ],
        "value": dbc.themes.CYBORG,
        "persistence": True,
    }
)
```
Its handled internally using the list `dark_themes = (dbc_dark_themes + custom_dark_themes)`

But if someone still applies `label_id` it won't be overridden, for instance it's possible to set `"label_id": ""` to remove the styling.